xdsl/frontend/onnx/context.py

```diff
@@ -0,0 +1,24 @@
+from onnx import ValueInfoProto
+
+from xdsl.frontend.onnx.type import get_type
+from xdsl.ir import Attribute
+
+
+class OnnxXdslMapping:
+    """The representation of the onnx context."""
+
+    type_by_name: dict[str, Attribute]
+
+    def __init__(self):
+        self.type_by_name = {}
+
+
+def visit_value_info(i: ValueInfoProto, ctx: OnnxXdslMapping) -> Attribute:
+    """Given the onnx ValueInforProto, it returns the corresponding Attribute stored in the context."""
+    name = i.name
+    t = get_type(i.type)
+    if name not in list(ctx.type_by_name.keys()):
+        ctx.type_by_name[name] = t
+        return t
+    else:
+        return ctx.type_by_name[name]
```

Comment:
```suggestion
    name = i.name
    if name in ctx.type_by_name:
        return ctx.type_by_name[name]
    t = get_type(i.type)
    ctx.type_by_name[name] = t
    return t
```
this is a pretty inefficient way to check that a key is in a dictionary, as lists have to be scanned linearly to test for whether they contain something, and dictionaries do this in constant time. Here is how I would recommend writing this. It has two advantages: we use the hash of the name to check whether we already have the information in O(1), and we only construct the type if it's not in the dictionary.

Reply:
Done